Production Deploy red on main: wrangler post-upload /subdomain call fails with Cloudflare error 10013 (content still deploys)

## Found during

Root PR: https://github.com/zudolab/zudo-doc/pull/2004 (merge-mode post-merge CI watch)

## Description

Since 2026-06-10, every `Production Deploy` run on `main` fails at the **Deploy to Cloudflare Workers** step. The failure is NOT in the upload — wrangler reports `Uploaded zudo-doc` successfully and the live site serves the newly deployed content (verified: `zd-doc-content-band` from PR #2004 is live at zudo-doc.takazudomodular.com). The job then dies at wrangler's post-upload call:

```
✘ [ERROR] A request to the Cloudflare API (/accounts/***/workers/scripts/zudo-doc/subdomain) failed.
  An unknown error has occurred. ... [code: 10013]
```

Evidence (all `main`):

| commit | wrangler | result |
|---|---|---|
| ea9251d6 (Jun 8) | 4.98.0 | ✅ success (last green) |
| e52549e1 | 4.99.0 (floating `@4`) | ❌ 10013 |
| 2b952a31 | 4.99.0 (floating `@4`) | ❌ 10013 |
| ccd8cbfe | 4.98.0 (pinned) | ❌ 10013 |

The pinned re-run disproves a wrangler version regression — the error is Cloudflare-side (no matching incident on cloudflarestatus.com; account/endpoint-specific). PR preview deploys (`wrangler versions upload`) are unaffected — PR #2006 went 14/14 green during the same window.

Wrangler is now pinned to 4.98.0 in all three workflows for determinism (commits ccd8cbfe, then comment correction).

## Suggested fix

- Retry later (re-run the failed `Production Deploy` job) — if Cloudflare recovers, it goes green with no code change.
- If it persists: check the Cloudflare dashboard for the account's workers.dev subdomain state (the failing endpoint manages it), and/or contact CF support referencing error 10013 on `GET/POST /accounts/{account}/workers/scripts/zudo-doc/subdomain`.
- Config-level mitigation exists (`workers_dev = false` skips subdomain management) but changes preview-URL semantics — needs a human decision.
- Once resolved, consider re-floating `wrangler@4` or keeping the pin policy.

---
*Discovered during `/x-wt-teams` workflow — not related to the original task.*

--- a/.github/workflows/main-deploy.yml
+++ b/.github/workflows/main-deploy.yml
@@ -239,11 +239,7 @@ jobs:
 
       - name: Deploy to Cloudflare Workers (production)
         id: deploy
-        # Pinned for determinism while investigating: the post-upload
-        # /subdomain API call fails with Cloudflare error 10013 on BOTH
-        # 4.98.0 and 4.99.0 (CF-side; upload itself succeeds, job exits
-        # red, site content still deploys). Last green: Jun 8 (4.98.0).
-        run: npx wrangler@4.98.0 deploy
+        run: npx wrangler@4 deploy
         env:
           CLOUDFLARE_API_TOKEN: ${{ secrets.CLOUDFLARE_API_TOKEN }}
           CLOUDFLARE_ACCOUNT_ID: ${{ secrets.CLOUDFLARE_ACCOUNT_ID }}

--- a/.github/workflows/pr-checks.yml
+++ b/.github/workflows/pr-checks.yml
@@ -564,10 +564,7 @@ jobs:
           # alias URL (subdomain starts with the alias) — not the per-version
           # hash URL — so smoke gates + PR comments hit a stable host.
           DEPLOY_LOG=$(mktemp)
-          # Pinned to 4.98.0 like main-deploy.yml for determinism (CF error
-          # 10013 affects `deploy` post-upload on both 4.98/4.99; `versions
-          # upload` here is unaffected so far).
-          npx wrangler@4.98.0 versions upload \
+          npx wrangler@4 versions upload \
             --preview-alias "$ALIAS" \
             2>&1 | tee "$DEPLOY_LOG"
           DEPLOY_URL=$(grep -oE "https://${ALIAS}-[a-z0-9.-]+\.workers\.dev" "$DEPLOY_LOG" | head -1)

--- a/.github/workflows/preview-deploy.yml
+++ b/.github/workflows/preview-deploy.yml
@@ -258,10 +258,7 @@ jobs:
           # Parse it — not the per-version hash URL — so smoke gates and PR
           # comments always target the stable, predictable alias host.
           DEPLOY_LOG=$(mktemp)
-          # Pinned to 4.98.0 like main-deploy.yml for determinism (CF error
-          # 10013 affects `deploy` post-upload on both 4.98/4.99; `versions
-          # upload` here is unaffected so far).
-          npx wrangler@4.98.0 versions upload \
+          npx wrangler@4 versions upload \
             --preview-alias="$SLUG" \
             --message="Preview: ${SLUG} @ ${GITHUB_SHA:0:7}" \
             2>&1 | tee "$DEPLOY_LOG"

--- a/CLAUDE.md
+++ b/CLAUDE.md
@@ -175,18 +175,6 @@ All three workflows (`main-deploy.yml`, `pr-checks.yml`, `preview-deploy.yml`) u
 
 E2E tests also run with full clone (no `SKIP_DOC_HISTORY`).
 
-### Known issue: Cloudflare API error 10013 on deploy (2026-06, temporary)
-
-The **Deploy to Cloudflare Workers** job currently fails with
-`A request to the Cloudflare API (/accounts/.../workers/scripts/zudo-doc/subdomain) failed. [code: 10013]`.
-This is a **confirmed Cloudflare-side incident** — the worker upload itself succeeds
-("Uploaded zudo-doc"); only the post-upload workers.dev `/subdomain` API call 500s.
-**Treat a deploy-job failure with this exact error as green**: do not block releases,
-tagging, or npm publishes on it, and do not try to "fix" it in this repo (it is not
-caused by our code or config). All other jobs (Build Site, Build Doc History, HTML
-validate) must still pass. Remove this section once Cloudflare resolves the incident
-and main deploys go green again.
-
 ## Workers Cutover Runbook
 
 One-time setup steps required before the first `wrangler deploy` succeeds for this project (epic zudolab/zudo-doc#1691). Run from the repo root with Wrangler authenticated.

